modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteMapSetOp.java

```diff
@@ -66,6 +68,32 @@ default RelDataType buildRowType() {
         return builder.build();
     }
 
+    /**
+     * Creates a row type produced by MAP phase of INTERSECT/EXCEPT operator.
+     * For input row (a:type1, b:type2) and {@code inputsNum} = {@code 3} it produces the following row type:
+     * <pre>
+     *     f0: type1
+     *     f1: type2
+     *     _count_0: int
+     *     _count_1: int
+     *     _count_2: int
+     * </pre>
+     */
+    public static RelDataType buildRowType(IgniteTypeFactory typeFactory, RelDataType rowType, int inputsNum) {
```

Comment:
Maybe this method should be moved to `PlanUtils` (as in PR with aggregates - https://github.com/apache/ignite-3/pull/2312). Alternatively,  it was a mistake moving type assembly related methods there in aggregates related PR, since other operators define helper functions closer to code where those functions are used.

Reply:
I like the idea to keep utility functions close to the place of usage

Reply:
I'd expect to see row layout in javadoc to AbstractSetOpNode.

Here, it just worth to mention that we append few `int` columns for aggregates needs.
However, you can left it as is.

Reply:
>I'd expect to see row layout in javadoc to AbstractSetOpNode.
Each operator has different structure for intermediate rows. There is a description of how both operator work subclasses of Grouping, that description includes row structure.